### ISSUE TITLE
Firefox: noscript tag escaping child elements on hydration

### ISSUE THREAD

**Issue by dimitri-lenkov** [solidjs/solid#1601]
### Describe the bug

Child elements within `<noscript>` are escaped on hydration causing a mis-match from the HTML spit-out from SSR.

### Your Example Website or App

Not Applicable

### Steps to Reproduce the Bug or Issue

1. Create a fresh SSR app.
2. Put a `<noscript>` tag with some elements inside.

```jsx
<noscript><strong>大丈夫</strong></noscript>
```

The browser (with JS enabled) will spit out:

```
Uncaught The browser resolved template HTML does not match JSX input:
<noscript>&lt;strong&gt;大丈夫&lt;/strong&gt;</noscript>

<noscript><strong>大丈夫</strong></noscript>. Is your HTML properly formed?
```

When JS is disabled, it works as expected.

### Expected behavior

I expect this to work with JS enabled.

### Platform

- OS: Linux
- Browser: Firefox
- Version: 110.0.1

**Comment by ryansolid** [comment]
I'm going to need a reproduction. Had no problem following the instructions: https://stackblitz.com/edit/github-mjp4mm?file=src%2Froot.tsx

**Comment by dimitri-lenkov** [comment]
Hi. I'll port my app to solid-start and see if that helps. Thanks.

**Comment by dimitri-lenkov** [comment]
Hello. I work in production, so I'm reluctant to go to a solid start at this point after reading the documentation. It looks great and I can't wait to try it in the future.

I went ahead and created a *very minimal* repository that mimics this problem:

https://github.com/dimitri-lenkov/solidjs-issue-1601

If there's anything I can do to help, please let me know. I look forward to replying to you and thank you for all the hard work you do!

**Comment by ryansolid** [comment]
Thank you for taking the time to make the reproduction. Hmm... it took me a moment to reproduce it as it doesn't occur in Chrome, Safari, or Edge. Only Firefox shows this message. Which suggests it is the only one that escapes `noscript`. On further inspection it only happens in template elements. So this is some quirky behavior of Firefox browser it seems where when setting `noscript` as the innerHTML of a template element it escapes its children. Tried to find an open Firefox issue but had no luck.

I've submitted a bug report: https://bugzilla.mozilla.org/show_bug.cgi?id=1821580

**Comment by dimitri-lenkov** [comment]
Interesting. Thank you for taking the time to confirm the bug & open a ticket on bugzilla. It seems like FF is strictly abiding by the HTML Standard:

- - -

> If scripting is enabled, [the noscript element is defined as containing only text](https://html.spec.whatwg.org/multipage/scripting.html#the-noscript-element) - though it must be parsable text, with some restrictions on content.

Source: https://stackoverflow.com/a/620955
Shog9 -- Mar 7, 2009 at 0:11

- - -

Peace :v:

**Comment by ryansolid** [comment]
I see.. so it's everyone else who has the bug. Well that's awkward. And sort of annoying because catering to both is going to be tricky.. because we can't know at compile time if the user has the JS on or off. As that needs different output.

**Comment by ryansolid** [comment]
Yeah Mozilla pointed out it was everyone else's bug. So we need to figure out solutions here. Now that we have better compile time validation would could remove the runtime validation. That is probably the path since I imagine there wouldn't be dynamic interpolation in the nocode section. I will see if I can get a fix in 1.7.
